homeassistant/components/ozw/node.py

```diff
@@ -0,0 +1,190 @@
+"""Node methods and classes for OpenZWave."""
+from openzwavemqtt.const import CommandClass, ValueType
+
+from homeassistant.components.websocket_api.const import (
+    ERR_NOT_FOUND,
+    ERR_NOT_SUPPORTED,
+)
+
+from .const import (
+    ATTR_CONFIG_PARAMETER,
+    ATTR_LABEL,
+    ATTR_MAX,
+    ATTR_MIN,
+    ATTR_OPTIONS,
+    ATTR_TYPE,
+    ATTR_VALUE,
+)
+
+
+class OZWValidationResponse:
+    """Class to hold response for validating an action."""
+
+    def __init__(self, success, *args, payload=None, err_type=None, err_msg=None):
+        """Initialize OZWValidationResponse."""
+        self.success = success
+        self.payload = payload
+        self.err_type = err_type
+        self.err_msg = err_msg
+        self.err_args = args
+
+    @staticmethod
+    def process_fail(err_type, err_msg, *args):
+        """Process an invalid request."""
+        return OZWValidationResponse(False, err_type=err_type, err_msg=err_msg, *args)
+
+    @staticmethod
+    def process_fail_on_type(value, new_value):
+        """Process an invalid request that fails type validation."""
+        return OZWValidationResponse.process_fail(
+            ERR_NOT_SUPPORTED,
+            "Configuration parameter type {} does not match the value type {}",
+            value.type,
+            type(new_value),
+        )
+
+    @staticmethod
+    def process_success(payload):
+        """Process a valid request."""
+        return OZWValidationResponse(True, payload=payload)
+
+
+def set_config_parameter(manager, instance_id, node_id, parameter_index, new_value):
```

Comment:
I wonder if this whole module belongs in the python-openzwave-mqtt library as a util. It doesn't contain anything specific for home assistant. It's just using the library and library data.

Reply:
That's not a bad idea. @cgarwood thoughts? I can submit a PR for this there if it makes sense

Reply:
Yeah I think this makes sense to move to the lib as it would be useful for others if they implemented the lib.